src/Dapr.Workflow/DaprWorkflowClientBuilderFactory.cs

```diff
@@ -0,0 +1,186 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using Grpc.Net.Client;
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Worker;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+#nullable enable
+
+namespace Dapr.Workflow;
+
+/// <summary>
+/// A factory for building a <see cref="DaprWorkflowClient"/>.
+/// </summary>
+internal sealed class DaprWorkflowClientBuilderFactory
+{
+    private readonly IConfiguration _configuration;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly IServiceCollection _services;
+    
+    /// <summary>
+    /// Constructor used to inject the required types into the factory.
+    /// </summary>
+    public DaprWorkflowClientBuilderFactory(IConfiguration configuration, IHttpClientFactory httpClientFactory, IServiceCollection services)
+    {
+        _configuration = configuration;
+        _httpClientFactory = httpClientFactory;
+        _services = services;
+    }
+    
+    /// <summary>
+    /// Responsible for building the client itself.
+    /// </summary>
+    /// <returns></returns>
+    public void CreateClientBuilder(Action<WorkflowRuntimeOptions> configure)
+    {
+        _services.AddDurableTaskClient(builder =>
+        {
+            var apiToken = GetApiToken(_configuration);
+            var grpcEndpoint = GetGrpcEndpoint(_configuration);
```

Comment:
What if `grpcEndpoint` is `null` or `""`?  That is accounted for for workers but not here.  Is there a difference?  I wonder if the same channel could be used for both client and worker.

Reply:
WIP - will use DaprDefaults as well